updates docker install docs (#1051)
fishtown-analytics/docs.getdbt.com: website/docs/dbt-cli/install/docker.md

description: "You can use Docker to install dbt and adapter plugins from the command line."
---

> You will find dbt Core and all adapter plugins maintained by dbt Labs available as [Docker](https://docs.docker.com/) images. We recommend you use Docker to install dbt in production because it includes dbt Core, one or more database adapters, and all dependencies. You can use Docker to install and develop locally if you don't have a python environment set up.

Reviewer: ```suggestion
 We recommend you use Docker to install dbt in production because it includes dbt Core, one or more database adapters, and all dependencies. You can find `dbt-core` and all dbt Labs-maintained database adapters available as Docker images hosted on our [GitHub packages page](https://github.com/orgs/dbt-labs/packages?visibility=public). Using Docker to install and develop locally does not require you to have a python environment set up. 

### Prerequisites
* You've installed Docker. For more information, see the [Docker](https://docs.docker.com/) site. 
* You understand which database adapter(s) you need. For more information, see [About dbt adapters](/dbt-cli/install/overview#about-dbt-adapters).
* You have a general understanding of the dbt, dbt workflow, developing locally in the command line interface (CLI). For more information, see [About dbt](/docs/introduction#how-do-i-use-dbt).
```

> dbt Core and all adapter plugins maintained by dbt Labs will are available as [Docker](https://docs.docker.com/) images. We recommend you use Docker to install dbt in production because it includes dbt Core, one or more database adapters, and all of their dependencies. You may also be able to use Docker to install and develop locally if you don't have a python environment set up.

Reviewer: ```suggestion
You will find dbt Core and all adapter plugins maintained by dbt Labs available as [Docker](https://docs.docker.com/) images. We recommend you use Docker to install dbt in production because it includes dbt Core, one or more database adapters, and all dependencies. You can use Docker to install and develop locally if you don't have a python environment set up.
```

Author: This all looks great to me, thanks for your proofreading/clarifying.

### Install a dbt Docker image from Github Packages

> Official dbt Docker images are hosted on our [GitHub packages page](https://github.com/orgs/dbt-labs/packages?visibility=public).  We maintain images and tags for each database adapter as well as a `latest` and `<major version>.<minor version>.latest` tag.  They can be installed by utilizing the `docker pull` command:

Reviewer: It seems like `latest` can be dropped, right?
```suggestion
Official dbt Docker images are hosted on our [GitHub packages page](https://github.com/orgs/dbt-labs/packages?visibility=public).  We maintain images and tags for each database adapter as well as a `latest` and `<major version>.<minor version>` tag.  They can be installed by utilizing the `docker pull` command:
```

Reviewer 2: @runleonarun We maintain two types of "latest" tags:
- Absolute `latest` (newest stable release)
- Major.Minor-specific `latest`, e.g. `1.0.latest`

As a concrete example: After we release v1.1.0, the `latest` tag will start installing v1.1. For users who are still on v1.0, the `1.0.latest` tag will still be available, and it will install the latest compatible patch (v1.0.x).

Some of this will be clearer once we have clearer version policies: #1061

Reviewer 2: @iknox-fa I do think we should clarify here when each `latest` tag is updated, especially for plugins. That's a question from a community member right now, since our plugin `latest` images haven't yet updated to install `dbt-core==1.0.2`: https://github.com/dbt-labs/dbt-core/issues/4756

Author: @jtcohen6 I assume you mean we should reference our current release schedule here?

Author: @runleonarun 
> It seems like latest can be dropped, right?

Nope, as @jtchoen6 points out

Reviewer 2: @iknox-fa Yeah! We've got a page now that we could link to: https://docs.getdbt.com/docs/core-versions

I think the question was: Are we rebuilding plugin images (e.g. `dbt-bigquery`) when new patch versions of `dbt-core` become available, so that they get access to the latest fixes from `dbt-core`? If so, do we update either the version tags or `latest` tags? If I understand right, we aren't doing this, since [`dbt-bigquery@1.0.latest`](https://github.com/dbt-labs/dbt-bigquery/pkgs/container/dbt-bigquery/17112343?tag=1.0.latest) hasn't been updated for a few months

> Offical dbt docker images are hosted on our [Github packages page](https://github.com/orgs/dbt-labs/packages?visibility=public).  We maintain images and tags for each database adapter as well as a `latest` and `<major version>.<minor version>.latest` tag.  They can be installed by utilizing the `docker pull` command:

Reviewer: ```suggestion
Official dbt Docker images are hosted on our [GitHub packages page](https://github.com/orgs/dbt-labs/packages?visibility=public).  We maintain images and tags for each database adapter as well as a `latest` and `<major version>.<minor version>.latest` tag.  They can be installed by utilizing the `docker pull` command:
```

```

> docker pull ghcr.io/dbt-labs/<db_adapter_name>:<version_tag>

Reviewer: I'd suggest using a real world example because we explain the break down in the previous paragraph.
```suggestion
docker pull ghcr.io/dbt-labs/dbt-core:latest
```

```

### Running a dbt Docker image in a container:

> The `ENTRYPOINT` for dbt Docker images is the command `dbt` so you can bind-mount your project to `/usr/app` and use dbt as normal:

Reviewer: @iknox-fa @jtcohen6 I got stuck here. I downloaded Docker and tried to copy/paste the following into a terminal window to no avail. Am I supposed to have a container running first? Does the code in the example have text I need to change? Thanks for your help! 🙇🏻‍♀️

Reviewer 2: @iknox-fa This references `my-dbt`, which seems to depend on having already built an image named "my-dbt" in the example below. Does that sound right? Would it make sense to re-order these?

Author: I changed it to a generic placeholder, which is what was intended.

Author: @runleonarun 
If you want to set up a time to walk through how to do this feel free to put something on my calendar.

Reviewer: Sounds great!

```
valid arg names for versioning are:
* `dbt_core_ref`
* `dbt_postgres_ref`
* `dbt_redshift_ref`
* `dbt_bigquery_ref`
* `dbt_snowflake_ref`
* `dbt_spark_ref`

> Note: Only overide a _single_ build arg for each build. Using multiple overides may lead to a non-functioning image.

If you wish to build an image with a third-party adapter you can use the `dbt-third-party` target.  This target requires you provide a path to the adapter that can be processed by `pip` by using the `dbt_third_party` build arg:
```
docker build --tag <your_image_name> \
  --target dbt-third-party \ 
  --build-arg dbt_third_party=<pip_parsable_install_string> \ 
  <path/to/dockerfile>
```

## Examples:
To build an image named "my-dbt" that supports redshift using the latest releases:
```
cd dbt-core/docker
docker build --tag my-dbt  --target dbt-redshift .
```

To build an image named "my-other-dbt" that supports bigquery using `dbt-core` version 0.21.latest and the bigquery adapter version 1.0.0b1:
```
cd dbt-core/docker
docker build \
  --tag my-other-dbt  \
  --target dbt-bigquery \
  --build-arg dbt_bigquery_ref=dbt-bigquery@v1.0.0b1 \
  --build-arg dbt_core_ref=dbt-core@0.21.latest  \
 .
```

> To build an image named "my-third-party-dbt" that uses [Materilize third party adapter](https://github.com/MaterializeInc/materialize/tree/main/misc/dbt-materialize) and the latest release of `dbt-core`:

Reviewer 2: We can link to our docs for using dbt + Materialize here:
```suggestion
To build an image named "my-third-party-dbt" that includes [the dbt adapter for Materialize](warehouse-profiles/materialize-profile) and the latest release of `dbt-core`:
```
